src/main/java/Jarvas.java

```diff
@@ -0,0 +1,104 @@
+                    return;
+                case "mark": {
+                    String index = userInput.substring(5); // obtain value after mark
+                    int i = Integer.parseInt(index) - 1; // convert to int
+                    Task markTask = tasks[i];
+
+                    tasks[i].setCompletedTrue();
+
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(markTask);
+                    break;
+                }
+                case "unmark": {
+                    String index = userInput.substring(7); // obtain value after unmark
+                    int i = Integer.parseInt(index) - 1; // convert to int
+                    Task markTask = tasks[i];
+
+                    tasks[i].setCompletedFalse();
+
+                    System.out.println("Okay, I've marked this task as not done yet:");
+                    System.out.println(markTask);
+                    break;
+                }
+                case "list":
+                    System.out.println("Here are the tasks in your list:");
+                    for (int i = 0; i < Task.enumerateTask(); i++) {
+                        System.out.println((i + 1) + ". "  + tasks[i]);
+                    }
+                    break;
+                case "todo": {
+                    int i = Task.enumerateTask();
+                    String label = userInput.substring(5).trim();
+                    ToDo toDo = new ToDo(label);
+
+
+                    tasks[i] = toDo;
+                    Reply.printReply(toDo, i+1);
+                    break;
+                }
+                case "event": {
+                    int i = Task.enumerateTask();
+                    String eventDetails = userInput.substring(6).trim();
+                    String[] eventParameters = Event.getInterval(eventDetails);
+                    String eventLabel = eventParameters[0];
+                    Event event = new Event(eventLabel, eventParameters[1], eventParameters[2]);
+
+                    tasks[i] = event;
+                    Reply.printReply(event, i+1);
+                    break;
+                }
+                case "deadline": {
+                    int i = Task.enumerateTask();
+                    String deadlineDetails = userInput.substring(9).trim();
+                    String[] deadlineParameters = Deadline.getDeadline(deadlineDetails);
+                    String deadlineLabel = deadlineParameters[0];
+                    Deadline deadline = new Deadline(deadlineLabel, deadlineParameters[1]);
+
+                    tasks[i] = deadline;
+                    Reply.printReply(deadline, i+1);
+                    break;
+                }
+                default:
+                    Reply.printReply("Invalid command provided");
+                }
+            } catch (NumberFormatException e) {
+                System.err.println("Invalid command format: " + e.getMessage());
+            }
```

Comment:
What edge cases does this try / catch block catch that make it necessary? Otherwise it is needlessly introducing an extra indentation to a significant chunk of code.